infrastructure/src/main/java/org/corfudb/infrastructure/LayoutServer.java

```diff
@@ -68,6 +70,7 @@
 //TODO Finer grained synchronization needed for this class.
 //TODO Need a janitor to cleanup old phases data and to fill up holes in layout history.
 @Slf4j
+@NotThreadSafe
```

Comment:
I believe that maithem made the class be single threaded execution to avoid the overhead in having all of its methods being synchronized. Thus the class would not be threadsafe since the methods are not monitored.